vignettes/metadata.Rmd

```diff
@@ -0,0 +1,102 @@
+---
+title: "Adding metadata"
+author: "Thierry Onkelinx"
+output:
+  rmarkdown::html_vignette:
+        fig_caption: yes
+vignette: >
+  %\VignetteIndexEntry{Adding metadata}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+## Introduction
+
+`git2rdata` supports extra metadata since version 0.4.1.
+Metadata is stored in a separate file with the same name as the data file, but with the extension `.yml`.
+The metadata file is a YAML file with a specific structure.
+The metadata file contains a generic section and a section for each variable in the data file.
+The generic section contains information about the data file as a whole.
+The variable sections contain information about the variables in the data file.
+The metadata file is stored in the same directory as the data file.
+
+The generic section contains the following mandatory fields, automatically created by `git2rdata`:
+
+- `git2rdata`: the version of `git2rdata` used to create the metadata.
+- `datahash`: the hash of the data file.
+- `hash`: the hash of the metadata file.
+- `optimize`: a logical indicating whether the data file is optimized for `git2rdata`.
```

Comment:
Nice work @ThierryO .
The only thing I noticed was that `optimize` and `optimized` are American English while I noticed that you prefer British English (e.g. behaviour instead of behavior).
Not sure.

No further remarks. It is clear to me.

Reply:
That reflects the argument name in [`write_vc()`](https://ropensci.github.io/git2rdata/reference/write_vc.html). Changing that would break at lot of code in the package and in the code of the package users.